pkg/client/api_client.go

```diff
@@ -13,12 +15,65 @@ import (
 type APIClient struct {
 	*RESTClient
 	SessionCookie *http.Cookie
+	ClientId      string
+	ClientSecret  string
 }
 
 const (
 	SessionCookie string = "JSESSIONID"
 )
 
+type HydraTokenBody struct {
+	AccessToken string `json:"access_token,omitempty"`
+	ExpiresIn   int    `json:"expires_in,omitempty"`
+	Scope       string `json:"scope,omitempty"`
+	TokenType   string `json:"token_type,omitempty"`
+}
+
+func (c *APIClient) GetJwtToken(hydraToken string) (string, error) {
+	if hydraToken == "" {
+		glog.V(4).Infof("The hydra token is empty")
+		return "", nil
+	}
+	request := c.Post().Resource(api.Resource_Type_auth_token).Header("x-oauth2", "hydra").Header("x-auth-token", hydraToken)
```

Comment:
Nit: A comment about the API or an href link to the API description will certainly be useful. Alternatively a link to some internal doc describing the overall flow I think will also be quite useful for future reference.